detection-rules/body_business_email_compromise_new_sender.yml

```diff
@@ -15,11 +15,34 @@ source: |
       // out of office auto-reply
       // the NLU model will handle these better natively soon
       or strings.istarts_with(subject.subject, "Automatic reply:")
-      or regex.imatch(subject.subject, '(\[[^\]]+\]\s?){0,3}(re|fwd?|automat.*)\s?:.*')
+      or regex.imatch(subject.subject,
+                      '(\[[^\]]+\]\s?){0,3}(re|fwd?|automat.*)\s?:.*'
+      )
     )
     and (
-      length(headers.references) > 0
-      or any(headers.hops, any(.fields, strings.ilike(.name, "In-Reply-To")))
+      (
+        length(headers.references) > 0
+        or any(headers.hops, any(.fields, strings.ilike(.name, "In-Reply-To")))
+      )
+      // catch fake message threads nested within legit replies
+      and not (
+        any([body.html.display_text, body.plain.raw],
+            3 of (
+              strings.icontains(., "from:"),
+              strings.icontains(., "to:"),
+              strings.icontains(., "sent:"),
+              strings.icontains(., "date:"),
+              strings.icontains(., "cc:"),
+              strings.icontains(., "subject:")
+            )
+        )
+        and (
+          // newly registered sender domain
+          network.whois(sender.email.domain).days_old <= 30
+          // or suspiciously long HTML div id
+          or regex.icontains(body.html.raw, '<div\sid="(.{200,})".*>')
```

Comment:
Just so i'm understanding this, this provides a mechanism to override the previous thread negation based on the sender domain again and a div length? 

If that's correct, maybe a new rule that specifically handles that case is warranted instead? 

This hunt reflects that logic, @peterdj45  let me know what you think.  https://platform.sublime.security/messages/hunt?huntId=019900fb-0d83-7ce0-8e5b-347f6b57b0c7